src/wazuh_modules/wm_syscollector.c

```diff
@@ -56,6 +70,11 @@ static void wm_sys_log_error(const char* log) {
 
 void* wm_sys_main(wm_sys_t *sys) 
 {
+    if (!sys->flags.enabled) {
```

Comment:
what will happen if the module is disabled and the destroy is called as part of the graceful shutdown? shutdown mutex will be used uninitialized.